src/os/ui/data/baseprovider.js

```diff
@@ -207,3 +220,33 @@ os.ui.data.BaseProvider.prototype.getUniqueId = function() {
 
   return id;
 };
+
+
+/**
+ * Gets the authentication info for the server (if any).
+ *
+ * @return {?os.auth.AuthEntry}
+ */
+os.ui.data.BaseProvider.prototype.getAuth = function() {
+  return os.auth.getAuth(this.getLabel());
+};
+
+
+/**
+ * @inheritDoc
+ */
+os.ui.data.BaseProvider.prototype.formatIcons = function() {
+  let icons = '';
+
+  const auth = this.getAuth();
+  if (auth) {
+    icons = `<i class="fa fa-sign-in" title="${auth.tooltip}"></i>`;
```

Comment:
I'm not certain we'll ever refactor all the old FA4 classes so we can stop using the shim, but ideally new code should use [FA5 classes](https://fontawesome.com/v5.15/icons/sign-in-alt?style=solid).